packages/dx-grid-core/src/plugins/table-column-resizing/helpers.ts

```diff
@@ -1,18 +1,64 @@
-import { ColumnSizeFn } from '../../types';
+import { ColumnSizeFn, ValidValueFn } from '../../types';
 
-export const getColumnSize: ColumnSizeFn = (
-  updatedColumn, {columnName, shift, minColumnWidth, maxColumnWidth, columnExtensions = [],
+export const getColumnsSizes: ColumnSizeFn = (
+  columnWidths, {
+    columnName, nextColumnName, nextColumnResizing, cachedWidths,
+    shift, minColumnWidth, maxColumnWidth, columnExtensions = [],
 }) => {
-  const extendedColumn = columnExtensions.find(elem => elem.columnName === columnName);
-  const minWidth = extendedColumn && extendedColumn.minWidth! >= 0
-    ? extendedColumn.minWidth
+  const column  = columnWidths.find(elem => elem.columnName === columnName)!;
+  const extension = columnExtensions.find(elem => elem.columnName === columnName);
+  const width = typeof column.width === 'number'
+    ? column.width
+    : cachedWidths[columnName];
+  const minWidth = extension && extension.minWidth! >= 0
+    ? extension.minWidth
     : minColumnWidth;
-  const maxWidth = extendedColumn && extendedColumn.maxWidth! >= 0
-    ? extendedColumn.maxWidth
+  const maxWidth = extension && extension.maxWidth! >= 0
+    ? extension.maxWidth
     : maxColumnWidth;
-  const size = Math.max(
+  let size = Math.max(
     minWidth!,
-    Math.min(updatedColumn.width! + shift, maxWidth!),
+    Math.min(width + shift, maxWidth!),
   );
-  return size;
+
+  if (nextColumnResizing) {
+    const nextColumn  = columnWidths.find(elem => elem.columnName === nextColumnName)!;
```

Comment:
These two parts of code, I mean a `column` and `nextColumn` looks very similar. Can we extract the same code?